src/main/java/org/aksw/simba/squirrel/frontier/Frontier.java

```diff
@@ -57,7 +58,7 @@
      * @param newUris
      *            the URIs that should be added to the {@link Frontier}
      */
-    void crawlingDone(List<CrawleableUri> crawledUris, List<CrawleableUri> newUris);
+    void crawlingDone(List<UriTimestampPair> crawledUris, List<CrawleableUri> newUris);
```

Comment:
I am not really happy with changing this interface since it causes changes in the `WorkerImpl` and maybe even the serialization as well... wouldn't it be easier to add the timestamp to the data of the crawlable URI? The Frontier could use this data flag to create the neessary UriTimestampPair and if it is not available, the Frontier can use its default values (e.g., recrawling timestamp = now + default time until recrawling).

But we can also discuss this. Maybe there is something that I simply overlook :smile:

Reply:
@MichaelRoeder you're right. We just realized that we do not need UriTimestampPair at all. I just embed the timestamp into CrawleableUri and then the code gets way simpler. I don't know why I did not see that 😀